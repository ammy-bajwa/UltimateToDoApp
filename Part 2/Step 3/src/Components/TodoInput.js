import React from "react";
import { withStyles } from "@material-ui/core/styles";
import Input from "@material-ui/core/Input";
import Button from "@material-ui/core/Button";
import Paper from "@material-ui/core/Paper";

const styles = theme => ({
  container: {
    display: "flex",
    flexWrap: "wrap",
    justifyContent: "center",
    alignItems: "center",
    flexDirection: "column",
    padding: "1rem 2.3rem"
  },
  input: {
    margin: "12px",
    "&:hover": {
      borderBottom: "1px solid rgb(241, 157, 96)"
    },
    "&:after": {
      borderBottom: "1px solid rgb(241, 157, 96)"
    },
    "&:before": {
      borderBottom: "1px solid rgb(241, 157, 96)"
    }
  },
  Button: {
    background: "linear-gradient(to bottom, #ffa562 0%, #ff875a 100%)",
    borderRadius: "50px",
    color: "#fff",
    padding: "20px 4rem 20px",
    fontSize: "21px"
  },
  paper: {
    borderRadius: "13px"
  }
});

class Inputs extends React.Component {
  state = {
    title: "",
    description: "",
    error: ""
  };

  _handleChange = name => event => {
    this.setState({
      [name]: event.target.value
    });
  };

  _handleSubmit = e => {
    e.preventDefault();
    if (!this.state.title || !this.state.description) {
      this.setState({ error: "Please fill both the feilds" });
    } else if (this.state.title && this.state.description) {
      const todo = {
        title: this.state.title,
        description: this.state.description
      };
      this.props.onSubmit(todo);
      this.setState({ error: "", title: "", description: "" });
      console.log(todo);
    }
  };
  render() {
    const { classes } = this.props;
    return (
      <Paper className={classes.paper}>
        <form onSubmit={this._handleSubmit} className={classes.container}>
          <Input
            fullWidth
            className={classes.input}
            value={this.state.title}
            onChange={this._handleChange("title")}
            inputProps={{ "aria-label": "Title" }}
            placeholder="Write the todo title"
          />
          <Input
            fullWidth
            className={classes.input}
            value={this.state.description}
            onChange={this._handleChange("description")}
            inputProps={{ "aria-label": "Description" }}
            placeholder="Write the description about what needs to be done"
            multiline
            rows="3"
          />
          {this.state.error && <p>{this.state.error}</p>}
          <Button color="secondary" type="submit" className={classes.Button}>
            Add todo
          </Button>
        </form>
      </Paper>
    );
  }
}

export default withStyles(styles)(Inputs);
