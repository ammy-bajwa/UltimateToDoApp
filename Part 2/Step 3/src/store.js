import { createStore, applyMiddleware } from "redux";
// import Reducer form reducers
import rootReducer from "./reducers";
// import our thunk middleware for redux
import thunk from "redux-thunk";

export const store = createStore(rootReducer, applyMiddleware(thunk));
