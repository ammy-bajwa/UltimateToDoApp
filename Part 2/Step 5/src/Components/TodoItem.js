import React from "react";
import { withStyles } from "@material-ui/core/styles";
import { connect } from "react-redux";
import Paper from "@material-ui/core/Paper";
import Button from "@material-ui/core/Button";
import Typography from "@material-ui/core/Typography";
import Grid from "@material-ui/core/Grid";
import FormGroup from "@material-ui/core/FormGroup";
import FormControlLabel from "@material-ui/core/FormControlLabel";
import Checkbox from "@material-ui/core/Checkbox";
import Dialog from "./Dialog";
import { deleteTodos, updateTodos } from "../actions";

const styles = theme => ({
  main: {
    ...theme.mixins.gutters(),
    paddingTop: theme.spacing.unit * 2,
    paddingBottom: theme.spacing.unit * 2
  },
  root: {
    color: "#f0925d",
    "&$checked": {
      color: "#f0925d"
    }
  },
  checked: {},
  todo: {
    marginTop: "1rem"
  },
  rootGrid: {
    maxWidth: "530px"
  },
  typographyHeader: {
    fontFamily: "Roboto",
    color: "#7a7a7a",
    fontWeight: "500",
    fontSize: "20px",
    textAlign: "start"
  },
  typographyPara: {
    fontFamily: "Roboto",
    color: "#7a7a7a",
    fontWeight: "400",
    fontSize: "16px",
    textAlign: "start"
  },
  button: {
    color: "#f0925d",
    fontSize: "14px",
    textTransform: "capitalize",
    background: "rgba(240, 142, 92, 0.08)"
  },
  label: {
    textTransform: "capitalize",
    color: "#ef865a",
    fontSize: "16px"
  },
  checkbox: {
    color: "green"
  },
  formLabel: {
    marginLeft: "15px",
    color: "#f0925d",
    "&$label": {
      color: "#f0925d"
    }
  },
  disabled: {
    textDecoration: "line-through"
  }
});

class TodoItem extends React.Component {
  state = {
    open: false
  };
  handleClickOpen = () => {
    this.setState({ open: true });
  };
  handleClose = () => {
    this.setState({ open: false });
  };
  handleDelete = () => {
    this.props.dispatch(deleteTodos(this.props.id));
    this.setState({ open: false });
  };
  handleCheck = () => {
    this.props.dispatch(updateTodos(!this.props.done, this.props.id));
  };
  render() {
    const props = this.props;
    const { classes } = props;
    return (
      <div className={classes.todo}>
        <Paper className={classes.main} style={{ borderRadius: "13px" }}>
          <Grid container className={classes.rootGrid}>
            <Grid item xs={8}>
              <Typography
                variant="headline"
                component="h3"
                className={classes.typographyHeader}
              >
                <span className={props.done && classes.disabled}>
                  {props.title}
                </span>
              </Typography>
              <Typography component="p" className={classes.typographyPara}>
                <span className={props.done && classes.disabled}>
                  {props.description}
                </span>
              </Typography>
            </Grid>
            <Grid item xs={4}>
              <Button className={classes.button} onClick={this.handleClickOpen}>
                Delete Todo
              </Button>
              <Dialog
                open={this.state.open}
                handleClose={this.handleClose}
                handleDelete={this.handleDelete}
              />
              <FormGroup>
                <FormControlLabel
                  className={classes.formLabel}
                  classes={{ label: classes.label, root: classes.root }}
                  control={
                    <Checkbox
                      checked={props.done}
                      value="checked"
                      classes={{ root: classes.root, checked: classes.checked }}
                      onChange={this.handleCheck}
                    />
                  }
                  label="Done"
                />
              </FormGroup>
            </Grid>
          </Grid>
        </Paper>
      </div>
    );
  }
}

export default connect()(withStyles(styles)(TodoItem));
