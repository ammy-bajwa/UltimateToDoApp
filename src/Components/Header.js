import React from "react";

const Header = props => {
  return <h1 className="app-logo-text">Ultimate Todo App</h1>;
};

export default Header;
