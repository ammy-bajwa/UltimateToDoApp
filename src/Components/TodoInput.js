import React from "react";
import { withStyles } from "@material-ui/core/styles";
import { Input, Button } from "@material-ui/core";

const styles = theme => ({
  container: {
    display: "flex",
    flexWrap: "wrap"
  },
  input: {
    margin: theme.spacing.unit
  }
});

class Inputs extends React.Component {
  state = {
    todo: ""
  };

  _handleChange = name => event => {
    this.setState({
      [name]: event.target.value
    });
  };

  _handleSubmit = e => {
    e.preventDefault();
    const todo = this.state.todo;
    this.props.onSubmit(todo);
    console.log(todo);
  };
  render() {
    const { classes } = this.props;
    return (
      <div className={classes.container}>
        <form onSubmit={this._handleSubmit}>
          <Input
            className={classes.input}
            value={this.state.todo}
            onChange={this._handleChange("todo")}
            inputProps={{ "aria-label": "Description" }}
          />
          <Button color="secondary" type="submit">
            Add todo
          </Button>
        </form>
      </div>
    );
  }
}

export default withStyles(styles)(Inputs);
