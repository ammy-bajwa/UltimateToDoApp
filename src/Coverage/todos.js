export default [
  {
    id: "1",
    title: "title",
    description: "some description",
    done: false
  },
  {
    id: "2",
    title: "title",
    description: "some description",
    done: false
  },
  {
    id: "3",
    title: "title",
    description: "some description",
    done: false
  }
];
