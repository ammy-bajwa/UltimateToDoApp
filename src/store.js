import { createStore, applyMiddleware } from "redux";
// import Reducer form reducers
import rootReducer from "./reducers";
// import our logger for redux
import { logger } from "redux-logger";
import thunk from "redux-thunk";

export const store = createStore(
  rootReducer,
  applyMiddleware(logger),
  applyMiddleware(thunk)
);
