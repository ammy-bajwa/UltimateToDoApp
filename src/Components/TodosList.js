import React from "react";
// import PropTypes from "prop-types";
import { withStyles } from "@material-ui/core/styles";
import TodoItem from "./TodoItem";
const styles = theme => ({
  root: {
    width: "100%"
  },
  disabed: {
    textDecoration: "line-through #000 dashed",
    color: "#505050"
  }
});

class CheckboxList extends React.Component {
  render() {
    const { classes } = this.props;

    return (
      <div className={classes.root}>
        {this.props.todos.map(item => (
          <div key={item.id}>
            <TodoItem
              title={item.title}
              description={item.description}
              done={item.done}
              id={item.id}
            />
          </div>
        ))}
      </div>
    );
  }
}

export default withStyles(styles)(CheckboxList);
