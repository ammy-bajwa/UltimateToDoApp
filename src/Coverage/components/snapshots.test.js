import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

// import Dialog from "../../Components/Dialog";
import Header from "../../Components/Header";
import TodoInput from "../../Components/TodoInput";
import TodoItem from "../../Components/TodoItem";
import TodosList from "../../Components/TodosList";
import todos from "../todos";
import { Provider } from "react-redux";
import { store } from "../../store";
import { Dialog } from "@material-ui/core";

describe("Snapshots of React Components", () => {
  it("Should Match Header Snapshot", () => {
    const wrapper = shallow(<Header />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it("Should Match TodoInput Snapshot", () => {
    const wrapper = shallow(<TodoInput />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it("Should Match TodoList  Snapshot", () => {
    const wrapper = shallow(<TodosList todos={todos} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it("Should Match TodoItem  Snapshot", () => {
    const wrapper = shallow(
      <Provider store={store}>
        <TodoItem
          title={todos[0].title}
          description={todos[0].description}
          done={todos[0].done}
          id={todos[0].id}
        />
      </Provider>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it("Should Match TodoItem  Snapshot", () => {
    const wrapper = shallow(
      <Provider store={store}>
        <Dialog open={false} />
      </Provider>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
