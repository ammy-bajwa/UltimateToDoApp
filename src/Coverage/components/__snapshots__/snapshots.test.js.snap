// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots of React Components Should Match Header Snapshot 1`] = `
<h1
  className="app-logo-text"
>
  Ultimate Todo App
</h1>
`;

exports[`Snapshots of React Components Should Match TodoInput Snapshot 1`] = `
<Inputs
  classes={
    Object {
      "Button": "Inputs-Button-03",
      "container": "Inputs-container-01",
      "input": "Inputs-input-02",
      "paper": "Inputs-paper-04",
    }
  }
/>
`;

exports[`Snapshots of React Components Should Match TodoItem  Snapshot 1`] = `
<Connect(WithStyles(TodoItem))
  description="some description"
  done={false}
  id="1"
  title="title"
/>
`;

exports[`Snapshots of React Components Should Match TodoItem  Snapshot 2`] = `
<WithStyles(Dialog)
  open={false}
/>
`;

exports[`Snapshots of React Components Should Match TodoList  Snapshot 1`] = `
<CheckboxList
  classes={
    Object {
      "disabed": "CheckboxList-disabed-06",
      "root": "CheckboxList-root-05",
    }
  }
  todos={
    Array [
      Object {
        "description": "some description",
        "done": false,
        "id": "1",
        "title": "title",
      },
      Object {
        "description": "some description",
        "done": false,
        "id": "2",
        "title": "title",
      },
      Object {
        "description": "some description",
        "done": false,
        "id": "3",
        "title": "title",
      },
    ]
  }
/>
`;
